### PR TITLE
Add start rollup job support to HL REST Client

### PR DESCRIPTION
**Pull request by cbuescher**
This change adds support for starting a rollup job to High Level REST Client.

Relates to #29827

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Review by hub-cap — CHANGES_REQUESTED** [review]
One nit and one larger comment that I think is worth discussing.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/AcknowledgedResponse.java [hunk 1]
```diff
@@ -58,7 +60,7 @@ public int hashCode() {
     public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
         builder.startObject();
         {
-            builder.field("acknowledged", isAcknowledged());
+            builder.field(field, isAcknowledged());
```
[the remark this answers is not in the document]
Comment (cbuescher) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/AcknowledgedResponse.java]: np, will update

**Comment by cbuescher** [comment]
@hub-cap thanks for the review, I updated the PR. Can you take a second look?

**Review by hub-cap — CHANGES_REQUESTED** [review]
Hey, awesome work. The only thing change is to remove the abstract from `getFieldName`, and just have the default AcknowledgeResponse impl of this method be "acknowledged", since thats the generic case that other APIs can and will use as well. 

I think it might be worth also moving the parser in. You can feel free to shoot this down, but i basically did the following to a sample response i had sitting in my current branch. Mind you, the tests started failing cuz i changed the parse field name, just to make sure it was working :)

```
public class DeleteRollupJobResponse extends AcknowledgedResponse {
    private static final String PARSE_FIELD_NAME = "some_other_ack";

    private static final ConstructingObjectParser<DeleteRollupJobResponse, Void> PARSER =
        AcknowledgedResponse.generateParser("delete_rollup_job_response", DeleteRollupJobResponse::new, PARSE_FIELD_NAME);

    public DeleteRollupJobResponse(boolean acknowledged) {
        super(acknowledged);
    }

    public static DeleteRollupJobResponse fromXContent(final XContentParser parser) throws IOException {
        return PARSER.parse(parser, null);
    }

    @Override
    String getFieldName() {
        return PARSE_FIELD_NAME;
    }
}
```

and added the following to AcknowledgedResponse
```
public static <T> ConstructingObjectParser generateParser(String name, Function<Boolean, T> ctor, String parseField) {
        ConstructingObjectParser p = new ConstructingObjectParser<>(name, true, args -> ctor.apply((boolean) args[0]));
        p.declareBoolean(constructorArg(), new ParseField(parseField));
        return p;
    }
```

**Comment by cbuescher** [comment]
@hub-cap I think I got what you mean. Please take a look at the last commit if thats the change you intended.

**Review by hub-cap — APPROVED** [review]
yea this is real nice now.

**Pull request merged**